src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -1591,6 +1592,57 @@ public string ToString([StringSyntax(StringSyntaxAttribute.NumericFormat)] strin
             return BigNumber.FormatBigInteger(this, format, NumberFormatInfo.GetInstance(provider));
         }
 
+        private string DebuggerDisplay
+        {
+            get
+            {
+                // For very big numbers, ToString can be too long or even timeout for Visual Studio to display
+                // Display an fast estimation value instead
+
+                // Use ToString for small values
+
+                if (_bits is null || _bits.Length <= 2)
+                {
+                    return ToString();
+                }
+
+                // Estimate the value x as L*2^n, while L is the value of high bits, and n is the length of low bits
+                // Represent L as k*10^i, then x = L*2^n = k*10^(i+n*Log10(2))
+                // Let m=n*Log10(2), the final result would be x = (k*10^(m-[m]))*10^(i+[m])
+
+                ulong highBits = ((ulong)_bits[^1] << kcbitUint) + _bits[^2];
+                double lowBitsCount32 = _bits.Length - 2; // if Length > int.MaxValue/32, counting in bits can cause overflow
+                double exponentLow = lowBitsCount32 * kcbitUint * Math.Log10(2);
+
+                // Max possible value of BigInteger is 2^(32*Array.MaxLength)-1 which is larger than 10^(2^35)
+                // Use long to avoid potential overflow
+                long exponent = (long)exponentLow;
+                double significand = (double)highBits * Math.Pow(10, exponentLow - exponent);
+
```

Comment:
See above, I believe the max possible size of BigInteger is smaller. @tannergooding can you confirm?

Regardless, even if I'm correct and you ease up on these restrictions, please use debug asserts to assert these assumptions.